### ISSUE TITLE
Graf Zeppelin + voice IDs

### ISSUE THREAD

**Issue by nonsensed** [KC3Kai/KC3Kai#1290]
Graf Zeppelin seems to break the ID format. Normally carriers will use ID 15 and 17 for their two different air battle quotes. The problem is that ID 17 is also used for night attacks, which carriers cannot normally do, so it works out. Since Graf CAN do night attacks, her two night battle subtitle IDs are a different number altogether. I tried each ID up to maybe 60 and none of them were attached to her night battle dialogues. I guess it's either not attached to an ID or a high number. I don't know how the IDs are mapped so I can't tell.

In short, I can't find the ID to attach Graf Zeppelin's night battle subtitles to.

Also, this affects the library because all carriers have "Yasen(2)" for their ID 17 instead of "Air Battle". "Battle" for carriers doubles as both their battle intro and one of their two different air battle quotes. So, the library could use a small update there, but it's really a minor issue.

**Comment by sinsinpub** [comment]
yea, i think all `Yasen(2)` should just be `Attack(2)` not only for carriers...
and I found Graf's yasen voice filename from cache is 917.mp3 (short), 918.mp3 (long)... no idea about its voiceline either

**Comment by dragonjet** [comment]
hm maybe that 917 and 918 is a legit `+1` and is a _copy_ of the _Attack(2)_ and not a swapped line. but i say this as a wild guess

**Comment by nonsensed** [comment]
Hmm. I tried 916, 917, 918, 919 as test IDs and didn't get anything in the quotes.json. If I have to do more than that to make it link, then I don't know how to do it.

Well, to illustrate, this is from testing in-game and also with the library.

Expected result:
17 Yasen Attack or Air Battle Attack (2)
18 Yasen Flagship Intro

Graf Result:
17 Air Battle Attack (2)
18 Air Battle Attack (2)

Her 17 and 18 are the exact same comment:
攻撃隊、出撃！Vorwärts! 
(Attack squadron, sortie! Forward!)

Graf's Yasen Flagship intro is:
フフフ…。夜戦か。いいだろう。このGraf Zeppelinがただの空母でないところを見せてやろう。艦隊、我に続け！追撃だ！
(Heheh... a night battle, is it? That's great. I will show you that I, Graf Zeppelin, am no ordinary carrier. Fleet, follow after me. Pursue!)

Graf's Yasen Attack is:
ただの空母でない所を見せてやろう。
(Let me show you I am no mere carrier.)

I tested in-game with Graf and she does say these during night battle. But none of the IDs from 1-59 will link to those night battle comments.

**Comment by dragonjet** [comment]
hmm could it be a wrong upload by the devs?

**Comment by LostLogia4** [comment]
This issue was idled for over a year, closing.

**Comment by nonsensed** [comment]
Graf Zeppelin's lines are still an active issue with no solution, however. The next time someone notices it, it'll come back.

**Comment by sinsinpub** [comment]
as i said before, kc devs use `917` (yasen attack), `918` (yasen opening) as special voice filename (also as they're greater than 53 which mean voice id keep the same without vckey conversation). i guess we can only also add them as special cases to solve Graf without any excuse.

scripts from decompiled swf, 432,353 is Graf (Kai):
```actionscript
// from BattleMan.swf/battle.phase.night.PhaseOpening._phaseConnection
         if(_loc1_ == 432 || _loc1_ == 353)
         {
            _loc2_ = 918;
         }
         Tween24.serial(Tween24.parallel(Tween24.waitEvent(_view.shutter,"ShutterView::EVENT_OPENED_COMPLETE"),Tween24.func(_view.shutter.open),Tween24.func(SoundUtil.playVoice,_loc1_,_loc2_))).onComplete(_phaseOpening).play();
```
```actionscript
// from BattleMan.swf/battle.phase.night.PhaseAttack._handlePlayVoiceInCutin
         if(_loc3_ is FriendShipData)
         {
            _loc2_ = _loc3_.getCharaID();
            _loc4_ = 17;
            if(_loc2_ == 432 || _loc2_ == 353)
            {
               _loc4_ = 917;
            }
            SoundUtil.playVoice(_loc2_,_loc4_);
```

**Comment by LostLogia4** [comment]
Is this one solved now or not? DO we still have any ongoing issues with this one?

**Comment by sinsinpub** [comment]
@LostLogia4 can be solved as long as TLors add Graf quotes at TL repo. i've only added JP ones.
